src/Marten/Events/EventMapping.cs

```diff
@@ -159,6 +159,14 @@ public T Resolve(DbDataReader reader, IIdentityMap map)
             return map.Get<T>(id, json);
         }
 
+        public async Task<T> ResolveAsync(DbDataReader reader, IIdentityMap map, CancellationToken token)
+        {
+            var id = await reader.GetFieldValueAsync<Guid>(0, token).ConfigureAwait(false);
+            var json = await reader.GetFieldValueAsync<string>(1, token).ConfigureAwait(false);
```

Comment:
It's not on purpose, just some sloppy inconsistency;-)

Event storage is its own thing that I wrote very early, and I just barely finished pulling it closer to how the documents get persisted.